numba/tests/test_array_manipulation.py

```diff
@@ -633,23 +629,40 @@ def test_squeeze_array_npm(self):
 
         self.assertIn("squeeze", str(raises.exception))
 
-    def test_add_axis2(self, flags=enable_pyobj_flags):
-        a = np.arange(9).reshape(3, 3)
+    def test_add_axis(self):
+        def np_new_axis_getitem(a, idx):
+            return a[idx]
+
+        def np_new_axis_setitem(a, idx, item):
+            a[idx] = item
+            return a
+
+        a = np.arange(4*5*6*7).reshape((4, 5, 6, 7))
+        idx_cases = [
+            (np.newaxis, slice(None)),
```

Comment:
Perhaps add some use cases where `np.newaxis` isn't the first item?

Reply:
Addressed in 89ba549